framework/src/base/FEProblem.C

```diff
@@ -2587,13 +2587,16 @@ FEProblem::computeUserObjects(ExecFlagType type/* = EXEC_TIMESTEP_END*/, UserObj
 }
 
 void
-FEProblem::reinitBecauseOfGhosting()
+FEProblem::reinitBecauseOfGhostingOrNewGeomObjects()
 {
-  // Need to see if _any_ processor has ghosted elems
-  dof_id_type ghosted = _ghosted_elems.size();
-  _communicator.sum(ghosted);
+  // Need to see if _any_ processor has ghosted elems or geometry objects.
+  bool needs_reinit = ! _ghosted_elems.empty();
+  needs_reinit = needs_reinit || ! _geometric_search_data._nearest_node_locators.empty();
+  needs_reinit = needs_reinit || ( _displaced_problem && ! _displaced_problem->geomSearchData()._nearest_node_locators.empty() );
+  dof_id_type do_reinit = static_cast<dof_id_type>(needs_reinit);
```

Comment:
My mistake; I merged two versions and forgot to clean up after myself. The
communicator.sum works fine with the bool.

On Tue, Jun 23, 2015 at 1:55 PM, Cody Permann <notifications@github.com>
wrote:

> In framework/src/base/FEProblem.C
> <https://github.com/idaholab/moose/pull/5265#discussion_r33084036>:
>
> >  {
> > -  // Need to see if _any_ processor has ghosted elems
> > -  dof_id_type ghosted = _ghosted_elems.size();
> > -  _communicator.sum(ghosted);
> > +  // Need to see if _any_ processor has ghosted elems or geometry objects.
> > +  bool needs_reinit = ! _ghosted_elems.empty();
> > +  needs_reinit = needs_reinit || ! _geometric_search_data._nearest_node_locators.empty();
> > +  needs_reinit = needs_reinit || ( _displaced_problem && ! _displaced_problem->geomSearchData()._nearest_node_locators.empty() );
> > +  dof_id_type do_reinit = static_cast<dof_id_type>(needs_reinit);
>
> @shanestafford <https://github.com/shanestafford>, not that it's a big
> deal but why cast this Boolean to an integer type only to use it as a
> Boolean again?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5265/files#r33084036>.
>